control-plane/p4RuntimeSerializer.cpp

```diff
@@ -639,6 +640,45 @@ getMatchType(cstring matchTypeName) {
     }
 }
 
+static int
+getTypeWidth(const IR::Type* type, TypeMap* typeMap) {
+    // p4runtime tests use bit Slices which are Type_Bits
+    // which will cause failure in minWidthBits.
+    // Thus compute width here for Slices and any other bit-field.
+    if (auto tb = type->to<IR::Type_Bits>()) {
+        return static_cast<int>(tb->width_bits());
+    }
+    auto ann = type->getAnnotation("p4runtime_translation");
+    if (ann != nullptr) {
+        auto sdnB = ann->expr[1]->to<IR::Constant>();
+        return static_cast<int>(sdnB->value.get_si());
+    }
+    return typeMap->minWidthBits(type, type->getNode());
+}
+
+/*
+ * The function returns a cstring for use as type_name for any nested type.
+*/
+static cstring
+getTypeName(const IR::Type* type, TypeMap* typeMap) {
+    cstring type_name = nullptr;
+    if (type == nullptr)
+        return type_name;
+
+    // p4runtime uses bit slices which are Type_Bits and cause
```

Comment:
This comment indicates that there is a bug somewhere else in the code. 
How does p4runtime use bit slices?

Reply:
As a MatchField.

Reply:
I don't see any other operations on bit slices in p4runtime.cpp.  The bit slice is merely used as a table key MatchField.